.github/workflows/project.yml

```diff
@@ -13,32 +13,40 @@ concurrency:
 
 jobs:
   integration:
+    strategy:
+      matrix:
+        version: [14, 16, 18]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: ./.github/actions/node/setup
       - run: yarn install
-      - uses: ./.github/actions/node/14
-      - run: yarn test:integration
-      - uses: ./.github/actions/node/16
-      - run: yarn test:integration
-      - uses: ./.github/actions/node/18
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.version }}
       - run: yarn test:integration
+
+  integration-latest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/node/setup
+      - run: yarn install
       - uses: ./.github/actions/node/latest
```

Comment:
I separate `integration-latest` into a different job so that we keep the definition of `latest` in the composite action